### PR TITLE
Added support for Reconciliation (CS1)

### PR DESCRIPTION
**Pull request by Armando-Ibarra**

### CODE REVIEW

File: lib/checkout_sdk/common/data_range_query_filter.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module CheckoutSdk
+  module Common
+    # @!attribute from
+    #   @return [DateTime]
+    # @!attribute to
+    #   @return [DateTime]
+    class DataRangeQueryFilter
```
Comment (martinseco) [inline lib/checkout_sdk/common/data_range_query_filter.rb:9]: I think there is a typo here. It should be `DateRangeQueryFilter` right?

File: lib/checkout_sdk/reconciliation/reconciliation_client.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        def initialize(api_client, configuration)
+          super(api_client, configuration, CheckoutSdk::AuthorizationType::SECRET_KEY)
+        end
+
+        # @param [ReconciliationQueryPaymentsFilter] payments_filter
+        def query_payments_report(payments_filter)
+          api_client.invoke_get(build_path(REPORTING, PAYMENTS), sdk_authorization, payments_filter)
+        end
+
+        def single_payment_report(payment_id)
```
Comment (martinseco) [inline lib/checkout_sdk/reconciliation/reconciliation_client.rb:24]: Missing documentation comment here

File: lib/checkout_sdk/api_client.rb [hunk 3]
```diff
@@ -127,11 +127,21 @@ def parse_response(response)
...
-      body = (JSON.parse(response.body, object_class: OpenStruct) if !response.body.nil? && response.body != '')
+      body = parse_json_or_contents(response)
       body = OpenStruct.new if body.nil?
       body = OpenStruct.new(items: body) if body.is_a? Array
       body.metadata = metadata if body.is_a? OpenStruct
       body
     end
+
+    def parse_json_or_contents(response)
+      return unless !response.body.nil? && response.body != ''
```
Comment (martinseco) [inline lib/checkout_sdk/api_client.rb:138]: I feel that this would read better like:
`return if response.body.nil? && response.body == ''`

Comment (martinseco) [reply lib/checkout_sdk/api_client.rb:138]: And now I'm in doubts if the && condition is ok there

**Review by martinseco — CHANGES_REQUESTED** [review]
Looks great! just some comments

File: lib/checkout_sdk/common/data_range_query_filter.rb (hunk 1 above)
Comment (Armando-Ibarra) [reply lib/checkout_sdk/common/data_range_query_filter.rb:9]: Ahh, took me a while to understand this, but yes it is `Date`

File: lib/checkout_sdk/api_client.rb (hunk 3 above)
Comment (Armando-Ibarra) [reply lib/checkout_sdk/api_client.rb:138]: That makes sense, there is no need to unless here, and also it should be an `or` as well, good catch

**Review by martinseco — APPROVED** [review]

**Pull request merged**